### PR TITLE
[ci skip] change all instances of blacklist and whitelist to denylist…

### PR DESCRIPTION
**Pull request by minaslater**
… and allowlist

### Summary

Per DHH, changed occurrence of "blacklist" to "denylist" and "whitelist" to "allowlist" in Rails documentation files. This pull request closes this [issue](https://github.com/rails/rails/issues/33677).

### Other Information

Adhering to information from the contribution guidelines re: documentation changes, only files in `rails/guides/source` were changed.

Finally, if your pull request affects documentation or any non-code
changes, guidelines for those changes are [available
here](http://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation)

Thanks for contributing to Rails!

### CODE REVIEW

**Review by justizin — APPROVED** [review]
There's no reason to believe they wouldn't, but a reading of this change shows that the grammar and code examples don't functionally change.

File: guides/source/form_helpers.md [hunk 1]
```diff
@@ -953,7 +953,7 @@ If the associated object is already saved, `fields_for` autogenerates a hidden i
 ### The Controller
 
 As usual you need to
-[whitelist the parameters](action_controller_overview.html#strong-parameters) in
+[allowlist the parameters](action_controller_overview.html#strong-parameters) in
```
Comment (rafaelfranca) [inline guides/source/form_helpers.md]: allow maybe?

**Review by rafaelfranca — COMMENTED** [review]
Thank you so much for working on this. I totally agree with the reasons for changing those terms.

I think we can improve most of those texts with different words than `allowedlist`. I made some comments with suggestions but I didn't got to review the whole patch with suggestion to avoid put a lot of comments.

Can you review the changes to make the text to look more fluid?

**Comment by minaslater** [comment]
@rafaelfranca Thanks for combing through for those awkward grammar. I've fixed all of them, mostly with your suggestion in the new commit.

File: guides/source/form_helpers.md [hunk 2]
```diff
@@ -999,7 +999,7 @@ remove addresses:
 <% end %>
 ```
 
-Don't forget to update the whitelisted params in your controller to also include
+Don't forget to update the permitted params in your controller to also include
```
Comment (huysentruitw) [inline guides/source/form_helpers.md]: By renaming whiltelist to:
* permitted
* allowed
* trusted

you're only creating more confusion.

Comment (tenderlove) [reply guides/source/form_helpers.md]: I think we could reword this to consistently use "permit" or "permitted".

File: guides/source/action_controller_overview.md [hunk 3]
```diff
@@ -193,7 +193,7 @@ In a given request, the method is not actually called for every single generated
 
 With strong parameters, Action Controller parameters are forbidden to
 be used in Active Model mass assignments until they have been
-whitelisted. This means that you'll have to make a conscious decision about
+allowed. This means that you'll have to make a conscious decision about
```
Comment (tenderlove) [inline guides/source/action_controller_overview.md]: This should probably be "_what is_ allowed" (or "_what is_ permitted")

File: guides/source/action_controller_overview.md [hunk 4]
```diff
@@ -241,7 +241,7 @@ Given
 params.permit(:id)
 ```
 
-the key `:id` will pass the whitelisting if it appears in `params` and
+the key `:id` will pass the allowlisting if it appears in `params` and
```
Comment (tenderlove) [inline guides/source/action_controller_overview.md]: > the key `:id` will be permitted if it appears in `params` and

or possibly

> the key `:id` will be permitted for inclusion if it appears in `params` and

File: guides/source/action_controller_overview.md [hunk 5]
```diff
@@ -349,7 +349,7 @@ end
 
 The strong parameter API was designed with the most common use cases
 in mind. It is not meant as a silver bullet to handle all of your
-whitelisting problems. However, you can easily mix the API with your
+allowlisting problems. However, you can easily mix the API with your
```
Comment (tenderlove) [inline guides/source/action_controller_overview.md]: > parameter filtering problems.

File: guides/source/getting_started.md [hunk 6]
```diff
@@ -779,10 +779,7 @@ extra fields with values that violated your application's integrity? They would
 be 'mass assigned' into your model and then into the database along with the
 good stuff - potentially breaking your application or worse.
 
-We have to whitelist our controller parameters to prevent wrongful mass
-assignment. In this case, we want to both allow and require the `title` and
-`text` parameters for valid use of `create`. The syntax for this introduces
-`require` and `permit`. The change will involve one line in the `create` action:
+We have to define our permitted controller parameters to prevent wrongful mass assignment. In this case, we want to both allow and require the `title` and `text` parameters for valid use of `create`. The syntax for this introduces `require` and `permit`. The change will involve one line in the `create` action:
```
Comment (tenderlove) [inline guides/source/getting_started.md]: I think this paragraph needs to be wrapped

File: guides/source/i18n.md [hunk 7]
```diff
@@ -77,8 +77,8 @@ There are also attribute readers and writers for the following attributes:
 load_path                 # Announce your custom translation files
 locale                    # Get and set the current locale
 default_locale            # Get and set the default locale
-available_locales         # Whitelist locales available for the application
-enforce_available_locales # Enforce locale whitelisting (true or false)
+available_locales         # Allowlist locales available for the application
```
Comment (tenderlove) [inline guides/source/i18n.md]: s/Allowlist/Permitted/

File: guides/source/i18n.md [hunk 8]
```diff
@@ -128,7 +128,7 @@ The load path must be specified before any translations are looked up. To change
 # Where the I18n library should search for translation files
 I18n.load_path += Dir[Rails.root.join('lib', 'locale', '*.{rb,yml}')]
 
-# Whitelist locales available for the application
+# Allowlist locales available for the application
```
Comment (tenderlove) [inline guides/source/i18n.md]: s/Allowlist/Permitted/

**Review by tenderlove — APPROVED** [review]
I added some comments.  I really like this change, but I think we should try to consistently use "permitted" (consistency should help people reading the docs, plus it shares the name of the method).  "Denylist" seems kind of awkward, but I'm having a hard time thinking of an antonym for "permit" besides "deny". 😊

File: guides/source/form_helpers.md (hunk 2 above)
Comment (minaslater) [reply guides/source/form_helpers.md]: @tenderlove Roger that.

**Comment by minaslater** [comment]
@tenderlove How do you feel about "restricted list" or "restrict" as an alternative?

**Comment by tenderlove** [comment]
> @tenderlove How do you feel about "restricted list" or "restrict" as an alternative?

Sounds good to me! 😄

**Review by guilleiguaran — APPROVED** [review]

**Review by rafaelfranca — APPROVED** [review]
